src/widgets/immunizations/immunizations-form.component.tsx

```diff
@@ -0,0 +1,348 @@
+                    data-testid="vaccinationExpirationInput"
+                    defaultValue={vaccinationExpiration}
+                    onChange={evt => setVaccinationExpiration(evt.target.value)}
+                  />
+                  <svg className="omrs-icon" role="img">
+                    <use xlinkHref="#omrs-icon-calendar"></use>
+                  </svg>
+                </div>
+              </div>
+              <div className={styles.immunizationsInputContainer}>
+                <label htmlFor="lotNumber">
+                  {t("lot number", "Lot Number")}
+                </label>
+                <div className="omrs-input-group">
+                  <input
+                    className="omrs-input-outlined"
+                    type="number"
+                    data-testid="lotNumberInput"
+                    style={{ height: "2.75rem" }}
+                    defaultValue={lotNumber}
+                    onChange={evt => setLotNumber(evt.target.value)}
+                  />
+                </div>
+              </div>
+              <div className={styles.immunizationsInputContainer}>
+                <label htmlFor="manufacturer">
+                  {t("manufacturer", "Manufacturer")}
+                </label>
+                <div className="omrs-input-group">
+                  <input
+                    className="omrs-input-outlined"
+                    type="text"
+                    data-testid="manufacturerInput"
+                    style={{ height: "2.75rem" }}
+                    defaultValue={manufacturer}
+                    onChange={evt => setManufacturer(evt.target.value)}
+                  />
+                </div>
+              </div>
+            </div>
+          </div>
+        </SummaryCard>
+        <div
+          className={
+            enableCreateButtons || enableEditButtons
+              ? `${styles.buttonStyles} ${styles.buttonStylesBorder}`
+              : styles.buttonStyles
+          }
+        >
+          <button
+            type="button"
+            className="omrs-btn omrs-outlined-neutral omrs-rounded"
+            style={{ width: "50%" }}
+            onClick={closeForm}
+          >
+            {t("cancel", "Cancel")}
+          </button>
+          <button
+            type="submit"
+            style={{ width: "50%" }}
+            className={
+              enableCreateButtons || enableEditButtons
+                ? "omrs-btn omrs-filled-action omrs-rounded"
+                : "omrs-btn omrs-outlined omrs-rounded"
+            }
+            disabled={viewEditForm ? !enableEditButtons : !enableCreateButtons}
+          >
+            {t("save", "Save")}
+          </button>
+        </div>
+      </form>
+    );
+  }
+
+  const closeForm = event => {
+    let userConfirmed: boolean = false;
+    if (formChanged) {
+      userConfirmed = confirm(
+        "There is ongoing work, are you sure you want to close this tab?"
```

Comment:
Well, I don't think we should use `confirm`, but that left aside we need a general way of dealing with that. Usually, you use a feature of (all known to me) routers; the ability to block transitions. Additionally, you'd need to hook on to the unload event to prevent closing the website.

We should have a convenience function (or hook) that we use for any form.

Independent of the technical implementation here I'd still ask for a translation of the string. Thanks!

Reply:
Ohh sure! I would add the translations for the confirm box.

Reply:
Regarding the something else instead of `confirm` box if there is an existing approach, we could adopt to that. We checked with few existing forms and they seems to be having `confirm` box only.